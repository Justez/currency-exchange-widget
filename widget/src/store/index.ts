import { createStore, applyMiddleware } from 'redux';
import createSagaMiddleware from 'redux-saga';
import { composeWithDevTools } from 'redux-devtools-extension/developmentOnly';

import createRootReducer from './root-reducer';
import sagas from './root-saga';

const configureStore = () => {
  const sagaMiddleware = createSagaMiddleware();

  const middlewares = [
    sagaMiddleware,
  ];

  const instance = createStore(
    createRootReducer(),
    composeWithDevTools(applyMiddleware(...middlewares)),
  );

  sagaMiddleware.run(sagas);

  return instance;
};

export type State = ReturnType<typeof store.getState>;

const store = configureStore();

export default store;
