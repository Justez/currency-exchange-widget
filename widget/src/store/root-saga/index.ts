export { default } from './root-saga';