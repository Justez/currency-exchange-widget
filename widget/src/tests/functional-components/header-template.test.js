import React from 'react';
import { Provider } from 'react-redux';
import renderer from 'react-test-renderer';

import store from 'store';
import Header from 'functional-components/header-template';

const component = (
    <Provider store={store}>
        <Header />
    </Provider>
);

describe('Header snapshots', () => {
    it('renders header', () => {
        const tree = renderer.create(component).toJSON();
        expect(tree).toMatchSnapshot();
    });
});