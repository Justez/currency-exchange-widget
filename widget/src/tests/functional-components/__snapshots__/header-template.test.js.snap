// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Header snapshots renders header 1`] = `
<div
  className="MuiBox-root MuiBox-root-3"
>
  <div
    className="MuiGrid-root MuiGrid-container MuiGrid-justify-xs-flex-end"
  >
    <div
      className="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-10"
    >
      <h6
        className="MuiTypography-root MuiTypography-h6"
      >
        Exchange currencies
      </h6>
    </div>
    <div
      className="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-1"
    >
      <button
        className="MuiButtonBase-root MuiIconButton-root makeStyles-iconButton-1"
        disabled={false}
        onBlur={[Function]}
        onClick={[Function]}
        onDragLeave={[Function]}
        onFocus={[Function]}
        onKeyDown={[Function]}
        onKeyUp={[Function]}
        onMouseDown={[Function]}
        onMouseLeave={[Function]}
        onMouseUp={[Function]}
        onTouchEnd={[Function]}
        onTouchMove={[Function]}
        onTouchStart={[Function]}
        tabIndex={0}
        type="button"
      >
        <span
          className="MuiIconButton-label"
        >
          <svg
            aria-hidden={true}
            className="MuiSvgIcon-root"
            focusable="false"
            style={
              Object {
                "height": "3vh",
                "marginTop": "-0.7vh",
              }
            }
            viewBox="0 0 24 24"
          >
            <path
              d="M18 7l-1.41-1.41-6.34 6.34 1.41 1.41L18 7zm4.24-1.41L11.66 16.17 7.48 12l-1.41 1.41L11.66 19l12-12-1.42-1.41zM.41 13.41L6 19l1.41-1.41L1.83 12 .41 13.41z"
            />
          </svg>
        </span>
      </button>
      <span
        className="MuiTypography-root makeStyles-label-2 MuiTypography-caption MuiTypography-noWrap"
      >
        place all
      </span>
    </div>
  </div>
</div>
`;
